packages/block-editor/src/store/actions.js

```diff
@@ -231,7 +254,13 @@ export function toggleSelection( isSelectionEnabled = true ) {
  */
 export function* replaceBlocks( clientIds, blocks, indexToSelect ) {
 	clientIds = castArray( clientIds );
-	blocks = castArray( blocks );
+	blocks = getBlocksWithAutoApplyStyles(
+		castArray( blocks ),
+		yield select(
+			'core/block-editor',
+			'getSettings',
+		)
+	);
```

Comment:
One question on my mind is whether we should log to the console any instances of auto-applying, so that anyone debugging their blocks can be aware of this transformation of `blocks` data.

Note that this is no substitute to a UI that is very clear in what is happening.

Reply:
We have some history in printing debugging information on the console for raw handling debugging. I'm not sure if for this case in concrete we should print a message, developers will soon know default styles can be changed by theme and the user so it will be something expected.